kusto/ingest/internal/resources/resources.go

```diff
@@ -63,12 +64,23 @@ func parse(uri string) (*URI, error) {
 		return nil, fmt.Errorf("error: Storage URI (%s) is invalid'", uri)
 	}
 
-	v := &URI{
-		u:          u,
-		account:    hostSplit[0],
-		objectType: hostSplit[1],
-		objectName: strings.TrimLeft(u.EscapedPath(), "/"),
-		sas:        u.Query(),
+	var v *URI
+	if len(hostSplit) == 5 {
```

Comment:
that's the only way to check? the zone can't happen to have 5 segments? is it what we do in other sdks?

Reply:
This is what we do for example- you see we use the objectName

	
	return azqueue.NewServiceURL(*service, p).NewQueueURL(queue.ObjectName()).NewMessagesURL(), nil

Reply:
https://github.com/Azure/azure-storage-blob-go is deprecated - see the readme (if that's what you meant)

I think they might have added methods to make it a bit easier recently.

Either way I'll resolve for now